File: Python3/프로그래머스/1/12940. 최대공약수와 최소공배수/최대공약수와 최소공배수.py
def solution(n, m):
    answer = [1, n * m]
    
    #최대공약수
    ans1 = min(n, m)
    while n % ans1 != 0 or m % ans1 != 0:
        ans1 -= 1
    
    #최소공배수
    ans2 = max(n, m)
    while ans2 % n != 0 or ans2 % m != 0:
        ans2 += 1
    
    return [ans1, ans2]

File: C++/프로그래머스/1/12940. 최대공약수와 최소공배수/최대공약수와 최소공배수.cpp
#include <vector>
#include <algorithm>

using namespace std;

vector<int> solution(int n, int m) {
    vector<int> answer;
    answer.push_back(1);
    answer.push_back(n * m);
    int ans1 = min(n, m);
    while(n % ans1 != 0 || m % ans1 != 0) {
        ans1--;
    }
    answer[0] = ans1;
    
    int ans2 = max(n, m);
    while(ans2 % n!= 0 || ans2 % m != 0) {
        ans2++;
    }
    answer[1] = ans2;
    
    return answer;
}